src/blind.cpp

```diff
@@ -36,6 +36,74 @@ class Blind_ECC_Init {
 
 static Blind_ECC_Init ecc_init_on_load;
 
+bool VerifyConfidentialPair(const CConfidentialValue& conf_value, const CConfidentialAsset& conf_asset, const CAmount& claimed_value, const CAsset& claimed_asset, const uint256& value_blinding_factor, const uint256& asset_blinding_factor) {
+    if (conf_value.IsNull() || conf_asset.IsNull() || claimed_asset.IsNull()) {
+        return false;
+    }
+
+    if (conf_value.IsExplicit()) {
+        // Match behavior of UnblindConfidentialPair
+        return false;
+    }
+    if (conf_asset.IsExplicit() && conf_asset.GetAsset() != claimed_asset) {
+        return false;
+    }
+
+    // Just to be safe
+    if (claimed_value > (int64_t)MAX_MONEY || !MoneyRange(claimed_value)) {
```

Comment:
claimed_value is already CAmount, and MoneyRange checks `nValue >= 0 && nValue <= MAX_MONEY`, is `claimed_value > (int64_t)MAX_MONEY` needed here ?

It would be needed if claimed_value would be `u_int64_t`, but I think here this extra check is not needed

Reply:
Hmm good question, I think I copied and pasted this check from elsewhere.

@stevenroose, thoughts? I will try to take a closer look tomorrow morning (my time).

Reply:
A similar code is at https://github.com/ElementsProject/elements/blob/ec9b28e0897c270bc551031741adb7361cec34b5/src/blind.cpp#L93-L95
but the difference is that `amount` there is of type uint64_t as received from  `secp256k1_rangeproof_rewind`. IIRC, this check is needed because unsigned to signed conversions when unsigned is outside of the signed range is implementation-defined (up until some recent C++ version, I don't remember which)

Reply:
Makes sense. The claimed value indeed comes from a CAmount from the unblinded tx, so I agree it doesn't need the extra MAX_MONEY check.

Reply:
Fixed.